pandas/core/arrays/datetimelike.py

```diff
@@ -461,10 +462,10 @@ def __getitem__(self, key):
     def __setitem__(
             self,
             key,    # type: Union[int, Sequence[int], Sequence[bool], slice]
-            value,  # type: Union[NaTType, Scalar, Sequence[Scalar]]
+            value,  # type: Union[NaTType, Any, Sequence[Any]]
```

Comment:
does Any not encompan NaTType here?

Reply:
Valid question. I mostly changed this to get it to pass since `Scalar` isn't a real thing as indicated by subsequent TODO. Think I should just remove Any altogether for the time being?

Reply:
no it would fail for that. What we need is a type that is an allowed datetime value, e.g: string, datetime / Timestamp, NaT). These can of course be done later, but should start setting up a pandas.typing i think.

Reply:
Yah, I'd rather have an import from pandas.typing and avoid imports that aren't "really" used (i.e. NaTType).